server/channels/api4/job_test.go

```diff
@@ -58,7 +61,11 @@ func TestGetJob(t *testing.T) {
 	_, err := th.App.Srv().Store().Job().Save(job)
 	require.NoError(t, err)
 
-	defer th.App.Srv().Store().Job().Delete(job.Id)
+	defer func() {
+		if result, appErr := th.App.Srv().Store().Job().Delete(job.Id); appErr != nil {
```

Comment:
Can you please remove the `if` check here and in the rest of the file and use `require.Nil` instead?

Reply:
I did mean this for the comment below. I did not get the `require.NoErrorF` to work. I got the following error: `channels/api4/job_test.go:41:12: undefined: require.NoErrorF (typecheck)`. Changed the line to: `require.NoError(t, appErr, "Failed to delete job (result: %v)", result)`. Not sure if this works though. It passed all test using `NoError` so i don't know

Reply:
Error: channels/api4/job_test.go:66:3: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:125:4: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:208:4: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:280:3: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:341:3: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:378:4: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:433:4: calling require.Nil on error, please use require.NoError instead

Reply:
My bad. `App.Srv().Store().Job().Delete` does return an `error`. So, `require.NoError` is indeed the right method to use.